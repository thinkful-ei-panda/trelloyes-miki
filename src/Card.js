import React from 'react';
import Delete from './Delete'
// import './Card.css';

function Card (props) {
    return (
        <div className="Card">
            <Delete id={props.id} title={props.title} handleDeleteCard={props.handleDeleteCard}/>
            <h3>{props.title}</h3>
            <p>{props.content}</p>
        </div>
    );   
};

export default Card;