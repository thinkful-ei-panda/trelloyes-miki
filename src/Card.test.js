import React from 'react';
import ReactDOM from 'react-dom';
import Card from './Card';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

describe('Card component', () => {
    // Smoke testing
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<Card />, div);
        ReactDOM.unmountComponentAtNode(div)
    });

    // Snapshot testing
    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<Card key="id" title="title" content="content" />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });

});