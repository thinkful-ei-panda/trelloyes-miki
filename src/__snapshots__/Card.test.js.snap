// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    onClick={[Function]}
  >
    - Delete Card
  </button>
  <h3>
    title
  </h3>
  <p>
    content
  </p>
</div>
`;
